cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -783,18 +785,25 @@ func ValidateUpgradeConfiguration(c *kubeadm.UpgradeConfiguration) field.ErrorLi
 	return allErrs
 }
 
-// ValidateCertValidity validates if the values for cert validity are too big
-func ValidateCertValidity(cfg *kubeadm.ClusterConfiguration) []error {
-	var allErrs []error
+// ValidateCertValidity validates if the values for cert validity are too big or not match
+func ValidateCertValidity(cfg *kubeadm.ClusterConfiguration) (warnings []error, errorList field.ErrorList) {
 	if cfg.CertificateValidityPeriod != nil && cfg.CertificateValidityPeriod.Duration > constants.CertificateValidityPeriod {
-		allErrs = append(allErrs,
+		warnings = append(warnings,
 			errors.Errorf("certificateValidityPeriod: the value %v is more than the recommended default for certificate expiration: %v",
 				cfg.CertificateValidityPeriod.Duration, constants.CertificateValidityPeriod))
 	}
 	if cfg.CACertificateValidityPeriod != nil && cfg.CACertificateValidityPeriod.Duration > constants.CACertificateValidityPeriod {
-		allErrs = append(allErrs,
+		warnings = append(warnings,
 			errors.Errorf("caCertificateValidityPeriod: the value %v is more than the recommended default for CA certificate expiration: %v",
 				cfg.CACertificateValidityPeriod.Duration, constants.CACertificateValidityPeriod))
 	}
-	return allErrs
+
+	if cfg.CertificateValidityPeriod != nil && cfg.CACertificateValidityPeriod != nil {
+		if cfg.CertificateValidityPeriod.Duration > cfg.CACertificateValidityPeriod.Duration {
+			errorList = append(errorList,
+				field.Invalid(field.NewPath("certificateValidityPeriod"), cfg.CertificateValidityPeriod.Duration, fmt.Sprintf("the value is greater than the caCertificateValidityPeriod (%v)", cfg.CACertificateValidityPeriod.Duration)))
```

Comment:
can we use errors.Errorf instead of field.Invalid for consistency with above?

Reply:
thanks, updated @neolit123

Reply:
Shall we merge it on 1.31?

Reply:
i think it can go in 1.31, given we are adding warnings only right?

Reply:
OK